src/rgw/rgw_rados.cc

```diff
@@ -1518,6 +1520,11 @@ int RGWZoneParams::fix_pool_names()
   return 0;
 }
 
+void RGWZoneParams::set_pool_namespaces()
+{
+  roles_pool.name_space = "role";
+}
+
```

Comment:
the namespaces should be parsed from the zone's pool strings, i.e. "pool:namespace". we shouldn't be hard-coding them anywhere